data/src/main/java/me/androidbox/data/remote/preference/PreferenceRepositoryImp.kt

```diff
@@ -0,0 +1,82 @@
+package me.androidbox.data.remote.preference
+
+
+import android.content.Context
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKey
+import dagger.hilt.android.qualifiers.ApplicationContext
+import me.androidbox.domain.authentication.model.LoginUser
+import me.androidbox.domain.authentication.preference.PreferenceRepository
+import javax.inject.Inject
+
+class PreferenceRepositoryImp @Inject constructor(
+    @ApplicationContext private val context: Context
+) : PreferenceRepository {
+
+    companion object {
+        const val TOKEN_KEY = "token_key"
+        const val USER_ID_KEY = "user_id_key"
+        const val FULL_NAME_KEY = "full_name"
+    }
+
+    private val masterKey by lazy {
+        MasterKey.Builder(context).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build()
+    }
+
+    private val sharedPreferences by lazy {
+        EncryptedSharedPreferences.create(
+            context,
+            "secret_shared_preferences",
+            masterKey,
+            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+        )
+    }
```

Comment:
What if you'd need these prefs somewhere else? You'd need to create a new instance (which I'm not sure how heavy that is for encrypted prefs). Instead, I'd prefer injecting it via Hilt

Reply:
@updated